File: FFCG.G4.Movies/FFCG.G4.Movies.Core/MovieCollection.cs
using System.Collections.Generic;

namespace FFCG.G4.Movies.Core
{
    public class MovieCollection
    {
        private readonly IStorage _storage;
        public IEnumerable<Movie> Movies { get { return _storage.All<Movie>(); } }

        public MovieCollection(IStorage storage)
        {
            _storage = storage;
        }

        public void AddMovie(Movie movie)
        {
            _storage.Store(movie);
        }
    }
}

File: FFCG.G7.Movies/src/Core/MovieCollection.ts
import { IStorage } from "./IStorage";
import { Movie } from "./Movie";

export class MovieCollection {
    private _storage: IStorage<Movie>;

    constructor(storage: IStorage<Movie>) {
        this._storage = storage
    }

    addMovie(movie: Movie) {
        this._storage.store(movie)
    }

    public get movies(): Array<Movie> {
        return this._storage.all();
    }
}
